### PR TITLE
Update Reactive Extensions to 3.1

### PR DESCRIPTION
**Pull request by ierof**
Update Reactive Extensions to 3.1

Changes proposed in this pull request:
 - Removing Reactive Extensions 2.2.5
 - Adding Reactive Extensions 3.1.1
 
How did I test this code:
 - Everything starts normally 

Has been tested on:
 - GIT 2.10 and above
 - Windows 7 and above

### CODE REVIEW

File: TranslationApp/app.config [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
       </dependentAssembly>
       <dependentAssembly>
-        <assemblyIdentity name="System.Reactive.Linq" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-2.2.5.0" newVersion="2.2.5.0"/>
+        <assemblyIdentity name="System.Reactive.Linq" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-2.2.5.0" newVersion="2.2.5.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="System.Reactive.Core" publicKeyToken="94bc3704cddfc263" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.0.3000.0" newVersion="3.0.3000.0" />
```
Comment (RussKie) [inline TranslationApp/app.config]: This looks odd - you have duplicated assembly bindings:

```
+        <assemblyIdentity name="System.Reactive.Core" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-2.2.5.0" newVersion="2.2.5.0" />

 +        <assemblyIdentity name="System.Reactive.Core" publicKeyToken="94bc3704cddfc263" culture="neutral" />
 +        <bindingRedirect oldVersion="0.0.0.0-3.0.3000.0" newVersion="3.0.3000.0" />
```

**Comment by vbjay** [comment]
Looks like the nuget uninstall script failed to cleanup.  Just delete the
bindings that have the 2.2.5 version numbers.

**Comment by ierof** [comment]
> Looks like the nuget uninstall script failed to cleanup.  Just delete the bindings that have the 2.2.5 version numbers.

Yes, it seems that NuGet uninstall script for Rx 2.2.5 do not remove own changes in app.congif. Cleaned them manually.

**Comment by RussKie** [comment]
Could you please check that projects dependent on RX are still operational.

For instance, with https://github.com/gitextensions/gitextensions/pull/3846 everything compiles, GE starts normally yet the TFS plugin fails with dependency issues.

**Comment by ierof** [comment]
Unfortunately I can not test the functionality of the TFS integration.
But it seems that everything works correctly.

I've clean everything, rebuild whole solution in release mode, opened each plug-in and Build server integration settings, then checked loaded assemblies. 
All plug-ins are loaded. And there are no GAC dependencies in Rx now.

s1

**Comment by RussKie** [comment]
@jbialobr any concerns?

**Pull request merged**